lib/maze/server.rb

```diff
@@ -35,26 +28,34 @@ class << self
       # @return [String] The UUID attached to all command requests for this session
       attr_reader :command_uuid
 
-      # The intended HTTP status code on a successful request
+      # Sets the status code generator for the HTTP verb given.  If no verb is given then the
+      # generator will be shared across all allowable HTTP verbs.
       #
-      # @return [Integer] The HTTP status code, defaults to 200
-      def status_code(verb=nil)
-        if @status_override_verb
-          override_status = @status_override_verb.eql?(verb)
+      # @param verb [String] HTTP verb
+      def set_status_code_generator(generator, verb = nil)
+        @generators ||= {}
+        if verb.nil?
```

Comment:
This can be way more Ruby:

```ruby
Array(verb || ALLOWED_HTTP_VERBS).each do |verb|
  @generators[verb] = generator
end
```

Reply:
Done (I will never make it as a Ruby style guide)